consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -1054,3 +1055,14 @@ func (x *XDPoS_v2) allowedToSend(chain consensus.ChainReader, blockHeader *types
 	}
 	return nil
 }
+
+// Periodlly execution(Attached to engine initialisation during "new"). Used for pool cleaning etc
+func (x *XDPoS_v2) periodicJob() {
+	go func() {
+		for {
+			<-time.After(utils.PeriodicJobPeriod * time.Second)
+			x.hygieneVotePool()
+			x.hygieneTimeoutPool()
```

Comment:
Need lock here?

Reply:
Why? they are cleaning up old rounds, it shouldn't need any locks as it's not an important operation.